schema/loader.go

```diff
@@ -0,0 +1,154 @@
+package schema
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"github.com/santhosh-tekuri/jsonschema/v5/httploader"
+)
+
+var (
+	//go:embed known_schemas
+	knownSchemas embed.FS
+)
+
+type (
+	SchemaFile string
+)
+
+const (
+	schemaDirectory = "known_schemas/"
+
+	// Presentation Exchange Schemas
+	PresentationDefinitionSchema              SchemaFile = "pe-presentation-definition.json"
+	PresentationDefinitionEnvelopeSchema      SchemaFile = "pe-presentation-definition-envelope.json"
+	PresentationSubmissionSchema              SchemaFile = "pe-presentation-submission.json"
+	PresentationClaimFormatDesignationsSchema SchemaFile = "pe-definition-claim-format-designations.json"
+	SubmissionClaimFormatDesignationsSchema   SchemaFile = "pe-submission-claim-format-designations.json"
+	SubmissionRequirementSchema               SchemaFile = "pe-submission-requirement.json"
+	SubmissionRequirementsSchema              SchemaFile = "pe-submission-requirements.json"
+
+	PresentationClaimFormatDesignationFormatDefinition SchemaFile = "pe-submission-claim-format-designations.json"
+
+	// Credential Manifest Schemas
+	CredentialManifestSchema    SchemaFile = "cm-credential-manifest.json"
+	CredentialApplicationSchema SchemaFile = "cm-credential-application.json"
+	CredentialResponseSchema    SchemaFile = "cm-credential-response.json"
+	OutputDescriptorsSchema     SchemaFile = "cm-output-descriptors.json"
+
+	// Wallet Rendering Schemas
+	DisplayMappingObjectSchema        SchemaFile = "wr-display-mapping-object.json"
+	EntityStylesSchema                SchemaFile = "wr-entity-styles.json"
+	LabeledDisplayMappingObjectSchema SchemaFile = "wr-labeled-display-mapping-object.json"
+
+	// VC JSON Schema Schemas
+	VerifiableCredentialJSONSchemaSchema SchemaFile = "vc-json-schema.json"
+)
+
+func (s SchemaFile) String() string {
+	return string(s)
+}
+
+// CachingLoader is a struct that holds local schemas
+type CachingLoader struct {
+	schemas map[string]string
+}
+
+// NewCachingLoader returns a new CachingLoader that enables the ability to cache http and https schemas
+func NewCachingLoader() CachingLoader {
+	localSchemas := make(map[string]string)
+	jsonschema.Loaders["https"] = func(url string) (io.ReadCloser, error) {
```

Comment:
I find it odd that a constructor has side effects. What's the rationale of patching `jsonschema` within the constructor?

Reply:
most, if not all, schemas will be loaded by http/s. having users configure additional loaders leaks details of the library that they shouldn't know or need to care about. this loader will work fine for the 99%